pkg/operator/backupcontroller/backupcontroller.go

```diff
@@ -0,0 +1,364 @@
+package backupcontroller
+
+import (
+	"context"
+	"fmt"
+	"github.com/openshift/cluster-etcd-operator/pkg/backuphelpers"
+	"github.com/openshift/library-go/pkg/operator/configobserver/featuregates"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sort"
+	"strings"
+	"time"
+
+	operatorv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+	operatorv1alpha1client "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1alpha1"
+	"github.com/openshift/cluster-etcd-operator/pkg/operator/etcd_assets"
+	"github.com/openshift/cluster-etcd-operator/pkg/operator/health"
+	"github.com/openshift/cluster-etcd-operator/pkg/operator/operatorclient"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apiserver/pkg/storage/names"
+	batchv1client "k8s.io/client-go/kubernetes/typed/batch/v1"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+const (
+	backupLabel      = "cluster-backup-job"
+	backupJobLabel   = "backup-name"
+	recentBackupPath = "/etc/kubernetes/cluster-backup"
+	backupDirEnvName = "CLUSTER_BACKUP_PATH"
+)
+
+type BackupController struct {
+	operatorClient      v1helpers.OperatorClient
+	backupsClient       operatorv1alpha1client.EtcdBackupsGetter
+	kubeClient          kubernetes.Interface
+	targetImagePullSpec string
+	featureGateAccessor featuregates.FeatureGateAccess
+}
+
+func NewBackupController(livenessChecker *health.MultiAlivenessChecker,
+	backupsClient operatorv1alpha1client.EtcdBackupsGetter,
+	kubeClient kubernetes.Interface,
+	eventRecorder events.Recorder,
+	targetImagePullSpec string,
+	accessor featuregates.FeatureGateAccess) factory.Controller {
+
+	c := &BackupController{
+		backupsClient:       backupsClient,
+		kubeClient:          kubeClient,
+		targetImagePullSpec: targetImagePullSpec,
+		featureGateAccessor: accessor,
+	}
+
+	syncer := health.NewDefaultCheckingSyncWrapper(c.sync)
+	livenessChecker.Add("BackupController", syncer)
+
+	return factory.New().ResyncEvery(1*time.Minute).
+		WithSync(syncer.Sync).ToController("BackupController", eventRecorder.WithComponentSuffix("backup-controller"))
+}
+
+func (c *BackupController) sync(ctx context.Context, _ factory.SyncContext) error {
+	if enabled, err := backuphelpers.AutoBackupFeatureGateEnabled(c.featureGateAccessor); !enabled {
+		if err != nil {
+			klog.V(4).Infof("BackupController error while checking feature flags: %v", err)
+		}
+		return nil
+	}
+
+	jobsClient := c.kubeClient.BatchV1().Jobs(operatorclient.TargetNamespace)
+	currentJobs, err := jobsClient.List(ctx, v1.ListOptions{LabelSelector: "app=" + backupLabel + ",state!=processed"})
```

Comment:
Interesting. So we have this label marker `state=processed` to filter out jobs already reconciled. I guess the conditions tell us that as well but I think we can't filter on those via ListOptions. So this can stay for now until we have something better in mind.